code/controllers/subsystem/goldmansachs.dm

```diff
@@ -0,0 +1,147 @@
+SUBSYSTEM_DEF(goldmansachs)
+	name = "Economy"
+	flags = SS_NO_INIT
+	var/paycheck_interval = 5 MINUTES
+	var/roundstart_paychecks = 5
+	var/budget_pool = 35000
+	var/list/department_accounts = list(ACCOUNT_CIV = ACCOUNT_CIV_NAME,
+										ACCOUNT_ENG = ACCOUNT_ENG_NAME,
+										ACCOUNT_SCI = ACCOUNT_SCI_NAME,
+										ACCOUNT_MED = ACCOUNT_MED_NAME,
+										ACCOUNT_SRV = ACCOUNT_SRV_NAME,
+										ACCOUNT_CAR = ACCOUNT_CAR_NAME,
+										ACCOUNT_SEC = ACCOUNT_SEC_NAME)
+	var/list/generated_accounts = list()
+	var/full_ancap = FALSE // Enables extra money charges for things that normally would be free, such as sleepers/cryo/cloning.
+							//Take care when enabling, as players will NOT respond well if the economy is set up for low cash flows.
+	var/datum/station_state/engineering_check = new /datum/station_state()
+	var/alive_humans_bounty = 100
+	var/crew_safety_bounty = 1500
+	var/monster_bounty = 150
+	var/mood_bounty = 100
+	var/techweb_bounty = 250
+	var/slime_bounty = list("grey" = 10,
+							// tier 1
+							"orange" = 100,
+							"metal" = 100,
+							"blue" = 100,
+							"purple" = 100,
+							// tier 2
+							"dark purple" = 500,
+							"dark blue" = 500,
+							"green" = 500,
+							"silver" = 500,
+							"gold" = 500,
+							"yellow" = 500,
+							"red" = 500,
+							"pink" = 500,
+							// tier 3
+							"cerulean" = 750,
+							"sepia" = 750,
+							"bluespace" = 750,
+							"pyrite" = 750,
+							"light pink" = 750,
+							"oil" = 750,
+							"adamantine" = 750,
+							// tier 4
+							"rainbow" = 1000)
+
+/datum/controller/subsystem/goldmansachs/fire(resumed = 0)
+	var/budget_to_hand_out = round(budget_pool / department_accounts.len)
+	for(var/A in department_accounts)
+		var/datum/bank_account/department/D = new /datum/bank_account/department(src)
+		D.department_id = A
+		D.account_holder = department_accounts[A]
+		D.account_balance = budget_to_hand_out
+		generated_accounts += D
+	for(var/A in GLOB.dep_cards)
+		var/obj/item/card/id/departmental_budget/C = A
+		var/datum/bank_account/B = get_dep_account(C.department_ID)
+		if(B)
+			C.registered_account = B
+			if(!B.bank_cards.Find(C))
+				B.bank_cards += C
+			C.name = "departmental card ([C.department_name])"
+			C.desc = "Provides access to the [C.department_name]."
+	addtimer(CALLBACK(src, .proc/its_payday_fellas), paycheck_interval)
+	flags |= SS_NO_FIRE
```

Comment:
Changing this flag requires an MC restart to take effect.

Also why are you making a subsystem that only fires once.

Reply:
I was told to use a subsystem for this since subsystems are the clean way to fire a thing off at roundstart. it also hold all the information about economy and runs the paychecks

Reply:
??? thats what Initialize() is for good you know that

Reply:
What? I thought Initialize was our New wrapper.